src/regexp.toit

```diff
@@ -1217,69 +1214,91 @@ class BackReference extends MiniExpAst:
     register_ = r
 
-  replace_with_ast ast/MiniExpAst -> none:
+  replace_with_ast ast/MiniExpAst_ -> none:
     ast_that_replaces_us_ = ast
 
-  generate compiler/MiniExpCompiler on_success/MiniExpLabel -> none:
+  generate compiler/MiniExpCompiler_ on_success/MiniExpLabel_ -> none:
     if register_ == null:
       compiler.generate ast_that_replaces_us_ on_success
     else:
       compiler.backtrack_on_back_reference_fail register_ compiler.case_sensitive
       compiler.goto on_success
 
-  analyze compiler/MiniExpCompiler -> MiniExpAnalysis:
+  analyze compiler/MiniExpCompiler_ -> MiniExpAnalysis_:
     compiler.add_back_reference this
-    return MiniExpAnalysis.know_nothing
+    return MiniExpAnalysis_.know_nothing
 
-class Capture extends MiniExpAst:
+class Capture_ extends MiniExpAst_:
   capture_count_ /int ::= ?
-  body_/MiniExpAst ::= ?
+  body_/MiniExpAst_ ::= ?
   start_register_/int? := null
   end_register_/int? := null
 
   constructor .capture_count_ .body_:
 
-  allocate_registers compiler/MiniExpCompiler -> none:
+  allocate_registers compiler/MiniExpCompiler_ -> none:
     if start_register_ == null:
       start_register_ = compiler.allocate_capture_registers
       end_register_ = start_register_ - 1
 
-  generate compiler/MiniExpCompiler on_success/MiniExpLabel -> none:
-    undo_start := MiniExpLabel
-    write_end := MiniExpLabel
-    undo_end := MiniExpLabel
-    compiler.copy_register start_register_ CURRENT_POSITION
+  generate compiler/MiniExpCompiler_ on_success/MiniExpLabel_ -> none:
+    undo_start := MiniExpLabel_
+    write_end := MiniExpLabel_
+    undo_end := MiniExpLabel_
+    compiler.copy_register start_register_ CURRENT_POSITION_
     compiler.push_backtrack undo_start
 
     compiler.generate body_ write_end
 
     compiler.bind write_end
-    compiler.copy_register end_register_ CURRENT_POSITION
+    compiler.copy_register end_register_ CURRENT_POSITION_
     compiler.push_backtrack undo_end
     compiler.goto on_success
 
     compiler.bind undo_start
-    compiler.copy_register start_register_ NO_POSITION_REGISTER
+    compiler.copy_register start_register_ NO_POSITION_REGISTER_
     compiler.backtrack
 
     compiler.bind undo_end
-    compiler.copy_register end_register_ NO_POSITION_REGISTER
+    compiler.copy_register end_register_ NO_POSITION_REGISTER_
     compiler.backtrack
 
-  analyze compiler/MiniExpCompiler -> MiniExpAnalysis:
+  analyze compiler/MiniExpCompiler_ -> MiniExpAnalysis_:
     allocate_registers compiler
     body_analysis := body_.analyze compiler
-    return MiniExpAnalysis.capture body_analysis start_register_ end_register_
+    return MiniExpAnalysis_.capture body_analysis start_register_ end_register_
 
 interface Match:
+  /// Returns the regular expression that created this match.
   pattern -> RegExp
+  /// Returns the input string that was matched by the regexp.
   input -> string
+  /// Returns the start index of the match in the string.
   start -> int
```

Comment:
Not a huge fan of the "start" name.
That sounds like a verb, but it isn't. Same for the other 'start' and the 'end'.
The `group` is harder, because it is both a verb and a noun.